lib/main.dart

```diff
@@ -12,6 +12,8 @@ import 'package:marquis_v2/providers/user.dart';
 import 'package:marquis_v2/router/route_information_parser.dart';
 import 'package:marquis_v2/router/router_delegate.dart';
 
+import 'services/snackbar_service.dart';
+
 void main() async {
```

Comment:
Since we modified the structure of the appstate now is different from `main`, better for us to add a check before opening the box for compatibility 
```
try {
    await Hive.openBox<AppStateData>("appState");
  } catch (e) {
    await Hive.deleteBoxFromDisk("appState");
    await Hive.openBox<AppStateData>("appState");
  }
```

Reply:
I used this suggestion to create a function  `_loadAppStateBox`. Calls to load the hive boxes have been switched to occur asynchronously. 

`await Future.wait([
    _loadAppStateBox(),
    Hive.openBox<UserData>("user"),
    Hive.openBox<LudoSessionData>("ludoSession"),
  ]);`